src/decode/xz.rs

```diff
@@ -361,7 +361,11 @@ where
             }
         }
         // TODO
-        CheckMethod::SHA256 => unimplemented!(),
+        CheckMethod::SHA256 => {
+            return Err(error::Error::XZError(
+                "unsupported SHA-256 checksum (not yet implemented)".to_string(),
```

Comment:
Nit: the other error messages start with an uppercase letter.

Reply:
Ah, good catch, aligned.
Though I usually see libraries not uppercasing, as these messages are usually wrapped by applications and printed like `app logic failed: library error details`, FWIW.

Reply:
The error interface should be improved anyway. See #6.